### PR TITLE
318-desktop swarm proposal

### PR DESCRIPTION
**Pull request by vkjr**
Desktop team swarm proposal by new template.
Research stage refined.

### CODE REVIEW

**Comment by vkjr** [comment]
@oskarth, per setup discussion initial idea for this swarm is to keep `react-native-deskop` library updated and create releases of Status app.

But releasing Status desktop involves tasks that currently fell out of responsibility of any team. For example - maintaining crash reporting tool. Also desktop team doesn't have Clojure developer assigned and that is important for polishing releases. 

@siphiuel and me were discussing that maybe it is a good idea to merge 2 teams - desktop and chat (@siphiuel and @yenda). And make this team responsible for Status desktop product (chat included).
During recent months, before new teams introduced, we were working exactly like this and @siphiuel acted as a lead of desktop project.

What do you think?

**Comment by oskarth** [comment]
I don't think it makes sense for desktop and chat team to be merged, as the priorities are different. If collaboration is needed then that's fine (that applies to Core as well cc @mandrigin), but ultimately the Desktop team should be responsible for releases.

Could you elaborate on the "maintaining crash reporting tool" and why this can't be done by the Desktop team?

cc @mandrigin @rachelhamlin

**Comment by rachelhamlin** [comment]
So to point out the obvious, desktop does not advance white paper completion and has loads more usability issues than mobile.

Are desktop issues considered relevant to this item from our proposal about Essential Swarms: `Onboarding/Patient Zero/Polish (Core/Relevant teams)` @oskarth?

Is desktop better off in maintenance mode with resources reallocated toward SNT utility and essential swarms?

Or does it continue to operate independently as a swarm with its own priorities, e.g. feature parity with mobile? 

My assumption is the latter, but it's not 100% clear and I think this is important to align on.

**Comment by vkjr** [comment]
@oskarth,
>Could you elaborate on the "maintaining crash reporting tool" and why this can't be done by the Desktop team?

This is for sure can and will be done by desktop team!

I'm trying to figure out the area of responsibility. And my confusion probably grows from misunderstanding of what is "creating releases". Originally I thought that it is about ensuring that Status Desktop can be build with `react-native-desktop` for all platforms and release builds work fine.
But if we define "creating releases" as all activity included in progressing Status desktop to align with mobile - then no questions from my side.

**Comment by oskarth** [comment]
Spoke briefly with @siphiuel about this.

> So to point out the obvious, desktop does not advance white paper completion and has loads more usability issues than mobile.

Agreed.

> Are desktop issues considered relevant to this item from our proposal about Essential Swarms: Onboarding/Patient Zero/Polish (Core/Relevant teams) @oskarth?

IMO this is largely focused on critical path for mobile. That said, if it takes 20% of the effort to get something reasonable for Desktop, I don't think this is a bad idea. Additionally, getting feature parity can be a bounty action. For chat/wallet/team/SNT swarms, this should be a product team decision, largely.

> Is desktop better off in maintenance mode with resources reallocated toward SNT utility and essential swarms?
>
> Or does it continue to operate independently as a swarm with its own priorities, e.g. feature parity with mobile? 

IMO it should operate independently with its own priorities, but with a slimmed down team (like a lot of other teams).

> This is for sure can and will be done by desktop team!

Great!

> I'm trying to figure out the area of responsibility. And my confusion probably grows from misunderstanding of what is "creating releases".

Agree. The most important thing is that it's clear who is responsible for what. Core (@mandrigin) is responsible for mobile releases. They explicitly marked Desktop as out of scope due to largely different stack and to avoid scope creep. So it would be useful figure out with them to make sure it is clear what the boundaries are.

---

In general, a lot of this are concerns that would come out of the Research phase, in conjunction with other stakeholders. So where Desktop is in 3 months should be a consideration informed by UXR (@hesterbruikman), and product.

@siphiuel can help out from Chat and Desktop POV in terms of his vision for how Desktop would progress.

@MaxRis this would be great to capture in this Desktop swarm as well, in terms of what the lowest hanging fruits are, what open problems are, and what the proposed work looks like for Q1 that helps Status achieve its stated goals top two prios - SNT utility and getting the app to a usable state.

Based on what comes out of this Research phase, we can titrate exact resource allocation, etc.

**Comment by vkjr** [comment]
On last call with @jarradh before changing the course we were discussing this desktop items:

- **upgrade react-native**
This one halfway done. 
PR with updated version already in `react-native-desktop`: status-im/react-native-desktop#422
PR for testing it with status-react - status-im/status-react#6983. Had issues but was postponed due to release
PR for mobile from @yenda in progress - c When this one merged it shouldn't take much time to merge others.

- **allow standard keyboard shortcuts**
Should be partially implemented in PR status-im/status-react#6951.
Probably can be postponed due to low priority.

- **native app & js communication**
Issue for this - status-im/status-react#6175
@MaxRis can elaborate more

- **publishing at platform's official marketplaces**
Probably can be postponed in favor of having more feature-full builds downloadable on our own site

- **better performance**
Related to memory issues - status-im/status-react#5271. Looks like they exist only when notifications enabled.

- **feature parity with mobile**
Wallet.
Has PR - status-im/status-react#6312 
@goranjovic, could you please elaborate, how many efforts would it take to make wallet available on desktop?

DApps browser.
As far as I know, no work done in this area. @siphiuel, @MaxRis, please correct me if I'm wrong.
Needs investigationю

- **windows desktop builds**
We already have windows builds on Jenkins.

- **windows. uwp support**
Probably can be postponed.


- **join relevant EthMagicians workshop**
Not sure I'm understanding this one. Probably can be postponed.

---

I would say that most important investigation to be done is how far we are from wallet, because it is tightly related to SNT utility. And what it would take to bring DApps.

@oskarth, @maxris, @siphiuel, wdyt?

**Comment by siphiuel** [comment]
@vkjr we are posting at the same time again :) Maybe we can discuss in https://notes.status.im/V3aRtupOTE-Kjfnw4e9AWA

File: ideas/311-desktop-team-proposal.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
+---
+id: 311-dekstop
+title: Desktop
+status: research/implementation
```
Comment (oskarth) [inline ideas/311-desktop-team-proposal.md]: research only probably for now - would be useful to sanity check with Rachel and Hester from Product and UX(R) POV

File: ideas/311-desktop-team-proposal.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  * Improve UI responsiveness
+ * **what should be done in `react-native-desktop` to make its maintenance easier**
+  * port to react-native 0.57. That will give us a possibility to make it an [out-of-tree](https://facebook.github.io/react-native/docs/out-of-tree-platforms) platform and facilitate future upgrades.
+    * PR with updated version [already in react-native-desktop](https://github.com/status-im/react-native-desktop/pull/422)
+    * [PR for testing it with status-react](https://github.com/status-im/status-react/pull/6983)
+    * PR for mobile [in progress](https://github.com/status-im/status-react/pull/6951)
+  * More documentation/tutorials for external contributors
+
+
+`Research output:`
```
Comment (oskarth) [inline ideas/311-desktop-team-proposal.md]: as well as bounties and user stories optimized for

Comment (vkjr) [reply ideas/311-desktop-team-proposal.md]: added

**Comment by jarradh** [comment]
This should be prioritized to focus on what can be done without changes that require a Go or Clojure resource as we cannot guarantee resources until the SNT utility cases and the mobile app polish is sufficient.

I suspect Core may want to jump in when it's time to integrate network incentivization.

Can we 'buy time' by handling the performance and memory issues first then get @siphiuel on it later once we're solid on other aspects of the software.

File: ideas/311-desktop-team-proposal.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
+---
+id: 311-dekstop
```
Comment (oskarth) [inline ideas/311-desktop-team-proposal.md]: will have to be changed, already taken

File: ideas/311-desktop-team-proposal.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+title: Desktop
+status: research
+lead-contributor: Vitaliy
```
Comment (oskarth) [inline ideas/311-desktop-team-proposal.md]: See comment below by J

File: ideas/311-desktop-team-proposal.md (hunk 3 above)
Comment (vkjr) [reply ideas/311-desktop-team-proposal.md]: done

File: ideas/311-desktop-team-proposal.md (hunk 4 above)
Comment (vkjr) [reply ideas/311-desktop-team-proposal.md]: done

**Comment by vkjr** [comment]
We will start prioritization on sync call to go as far as possible without Clojure/Go resources

**Pull request merged**